src/components/DetailsStatistics/DetailsStatistics.js

```diff
@@ -17,31 +17,46 @@ illegal under applicable law, and the grant of the foregoing license
 under the Apache 2.0 license is conditioned upon your compliance with
 such restriction.
 */
-import React, { useMemo } from 'react'
+import React, { useMemo, memo } from 'react'
 import PropTypes from 'prop-types'
 import classnames from 'classnames'
 
-import MlChart from '../../common/Chart/MlChart'
 import { Tooltip, TextTooltipTemplate } from 'igz-controls/components'
+import DetailsStatisticsTableRow from './DetailsStatisticsTableRow'
 
-import { generateStatistics } from './detailsStatistics.util'
+import { DETAILS_STATISTICS_TABLE_BODY_ID, DETAILS_STATISTICS_TABLE_ID, generateStatistics } from './detailsStatistics.util'
 import { getHistogramChartConfig } from '../../utils/getHistogramChartConfig'
+import { isRowRendered, useVirtualization } from '../../hooks/useVirtualization.hook'
 
+import cssVariables from './detailsStatistics.scss'
 import './detailsStatistics.scss'
-import colors from 'igz-controls/scss/colors.scss'
+
+
 
 const DetailsStatistics = ({ selectedItem }) => {
-  const statistics = generateStatistics(selectedItem)
+  const statistics = useMemo(() => generateStatistics(selectedItem), [selectedItem])
+  const rowsSizes = useMemo(() => new Array(statistics.length).fill(parseInt(cssVariables.detailsStatisticsRowHeight)), [statistics])
+  const heightData = useMemo(() => ({headerRowHeight: cssVariables.detailsStatisticsHeaderRowHeight}), [])
   const chartConfig = useMemo(getHistogramChartConfig, [])
-  const headers = Object.entries(statistics[0]).map(([label, value]) => ({
-    label,
-    type: value.type,
-    hidden: statistics.every(statisticsItem => statisticsItem[label].hidden)
-  }))
+  const headers = useMemo(
+    () =>
+      Object.entries(statistics[0]).map(([label, value]) => ({
+        label,
+        type: value.type,
+        hidden: statistics.every(statisticsItem => statisticsItem[label].hidden)
+      })),
+    [statistics]
+  )
+  const virtualizationConfig = useVirtualization({
+    rowsSizes,
+    tableId: DETAILS_STATISTICS_TABLE_ID,
+    tableBodyId: DETAILS_STATISTICS_TABLE_BODY_ID,
+    heightData
+  })
 
   return (
     <div className="details-statistics">
-      <div className="details-statistics__table">
+      <div className="details-statistics__table lazy-table" id={DETAILS_STATISTICS_TABLE_ID}>
```

Comment:
move class to details-statistics try to use constant and rename class to general one